src/libertem/udf/stddev.py

```diff
@@ -161,26 +256,30 @@ def process_tile(self, tile):
         if key not in self.results.num_frame[0]:
             self.results.num_frame[0][key] = 0
 
-        tile_sum = tile.sum(axis=0)
+        N0 = self.results.num_frame[0][key]
+        N1 = tile.shape[0]
 
-        p0 = VariancePart(
-            var=self.results.var,
-            sum_im=self.results.sum_frame,
-            N=self.results.num_frame[0][key]
-        )
-        p1 = VariancePart(
-            # We doctor ddof to ensure the sum of variances is divided by one.
-            # That way we avoid multiplying by N again for this algorithm
-            var=np.var(tile, axis=0, ddof=tile.shape[0]-1),
-            sum_im=tile_sum,
-            N=tile.shape[0]
-        )
-        compute_merge = merge(p0, p1)
+        oldshape = self.results.sum_frame.shape
 
-        self.results.var[:] = compute_merge.var
+        tile.shape = (N1, -1)
 
-        self.results.sum_frame[:] = compute_merge.sum_im
-        self.results.num_frame[0][key] = compute_merge.N
+        self.results.sum_frame.shape = (-1, )
+        self.results.var.shape = (-1, )
+
+        if N0 == 0:
+            self.results.sum_frame[:] = tile.sum(axis=0)
+            self.results.var[:] = np.var(tile, axis=0, ddof=N1-1)
```

Comment:
Maybe the `ddof` trickery does merit a comment, like in the old version, but also in the buffer description, because here, `self.results.var` is actually `N * Var(X)`

Reply:
I've renamed all instances of `var` to `varsum` where it contains the sum of variances and not the variance.

Comment for ddof added!